tt_metal/hw/inc/compile_time_args.h

```diff
@@ -4,6 +4,35 @@
 
 #pragma once
 
-namespace ckernel {
-#define get_compile_time_arg_val(arg_idx) KERNEL_COMPILE_TIME_ARG_##arg_idx
-}  // namespace ckernel
+#include <array>
+
+template <class T, class... Ts>
+FORCE_INLINE constexpr std::array<T, sizeof...(Ts)> make_array(Ts... values) {
+    return {T(values)...};
+}
+
+#if defined(KERNEL_COMPILE_TIME_ARGS)
+constexpr auto kernel_compile_time_args = make_array<std::uint32_t>(KERNEL_COMPILE_TIME_ARGS);
+#else
+constexpr auto kernel_compile_time_args = make_array<std::uint32_t>();
+#endif
```

Comment:
```suggestion
#ifndef KERNEL_COMPILE_TIME_ARGS
#define KERNEL_COMPILE_TIME_ARGS
#endif

constexpr auto kernel_compile_time_args = make_array<std::uint32_t>(KERNEL_COMPILE_TIME_ARGS);
```

Reply:
You could also just not do the early return in `process_compile_time_args` and unconditionally define `-DKERNEL_COMPILE_TIME_ARGS`